### PR TITLE
Feature/add jupyter notebook support

### PR DESCRIPTION
**Pull request by duysqubix**
#### Brief overview of PR changes/additions
Adds support for jupyter notebook via `evennia shell_plus --notebook`

#### Motivation for adding to Evennia

I like that you can use bpython or ipython interactively within your game, but it is frustrating at times
to have to redo everything from scratch to make sure your new feature or changed actually happened.

I like the idea of having a persistent code storage for testing and/sandbox environment, as well as 
more easily execute blocks of code.

### CODE REVIEW

**Comment by duysqubix** [comment]
hmm so besides the errors that sprung up from the checks...

It seems that `evennia shell_plus` works and you can interact with the database as well as the game..

but doing `evennia shell_plus --notebook` does start a jupyter instance, but it seems variables aren't initialized

**Comment by duysqubix** [comment]
alright.. so still investigating this, but i got the notebook to initialize properly by

```python
import evennia
evennia._init()
```

I am not sure why it doesn't do this automatically. Since the 'shell_plus' is found [here](https://github.com/duysqubix/evennia/blob/92a659da6c1eedab98e74ddd48cca51acfca2e40/evennia/server/evennia_launcher.py#L2262). Which sets `check_db == True`

**Comment by duysqubix** [comment]
Alright, so it seems that it does in fact call `evennia._init()` but the results of that doesn't get relayed to the notebook properly..

File: requirements.txt [hunk 1]
```diff
@@ -8,9 +8,12 @@ pytz
 djangorestframework >= 3.10.3, < 3.12
 django-filter >= 2.2.0, < 2.3
 django-sekizai
+django-extensions >= 3.1.0
 inflect
 autobahn >= 17.9.3
 lunr == 0.5.6
+ipython >= 7.19.0
+jupyter == 1.0.0
```
Comment (Griatch) [inline requirements.txt:16]: While I can see a point for having `ipython` in `requirements.txt`, the other two are too specialized. Move these to `requirements_extra.txt` instead - this is used for optional requirements like this.

**Review by Griatch — CHANGES_REQUESTED** [review]
Looking good, I have one request for change in the requirements.

File: requirements.txt (hunk 1 above)
Comment (duysqubix) [reply requirements.txt:16]: Alright. I will move those requirements to the extra file. 

So this works perfectly on my end, in fact I am using the notebooks to fine-tune my leveling algorithm. 

A couple of caveats and would need to be documented (not sure where) is 1) that you have to run ‘evennia shell_plus —notebook’ in the root of your game folder.  

2) when you create a new notebook you must run

‘’’python
import evennia
evennia._init()
‘’’

Comment (Griatch) [reply requirements.txt:16]: For now you can add your info as a new section below this one: https://github.com/evennia/evennia/blob/develop/docs/source/Coding/Coding-Introduction.md#explore-evennia-interactively . You don't have to set up/build the docs locally if you don't want to, just add the text to the file and I'll rebuild next time I update them.

**Comment by duysqubix** [comment]
Made the requested changes.

**Pull request merged**